storage/remote/queue_manager.go

```diff
@@ -375,12 +398,46 @@ func (t *QueueManager) Start() {
 	go t.reshardLoop()
 }
 
+// loadCheck will check if the SegmentRecord.json contents and segment is valid.
+func (t *QueueManager) loadCheck() bool {
+	var check CheckpointRecord
+
+	jsonFile, err := os.Open("data/CheckpointRecord.json")
+	defer jsonFile.Close()
+
+	byteValue, _ := ioutil.ReadAll(jsonFile)
+	err = json.Unmarshal(byteValue, &check)
+
+	if err != nil {
+		level.Error(t.logger).Log("err", err)
+		return false
+	}
+
+	for _, record := range check.Checkpoints {
+		segmentNumber, err := strconv.Atoi(record.Segment)
+
+		if err != nil {
+			level.Error(t.logger).Log("err", err)
+			return false
+		}
+
+		if segmentNumber <= wal.MinSegmentNumber || segmentNumber >= wal.MaxSegmentNumber {
```

Comment:
this check still feels strange to me given that we don't already set a min or max segment # already, I think `if segmentNumber >= 0` we should consider it valid, whether the file exists is another issue

Reply:
I think that's a typo i made. It should be `if segmentNumber < wal.MinSegmentNumber || segmentNumber > wal.MaxSegmentNumber` return false. Min number being 0 and max number being 99999999